pkg/io/binaryReader.go

```diff
@@ -59,6 +56,38 @@ func (r *BinReader) ReadArray(t interface{}) interface{} {
 	return arr.Interface()
 }
 
+// ReadArrayTo reads array into value which must be
+// a pointer to a slice.
+func (r *BinReader) ReadArrayTo(t interface{}) {
```

Comment:
Do we really need `ReadArray` with this?

Reply:
We can get rid of it but then in case of slice of pointers perform allocation for every element.

Reply:
`ReadArray` now accepts a pointer to slice.
A drawback is that now it cannot unmarshal an array of interfaces. Anyway have not encountered any of them.